cmake/modules/AddSwift.cmake

```diff
@@ -92,9 +92,18 @@ function(_add_host_variant_c_compile_link_flags name)
       MACCATALYST_BUILD_FLAVOR ""
       DEPLOYMENT_VERSION "${DEPLOYMENT_VERSION}")
     target_compile_options(${name} PRIVATE $<$<COMPILE_LANGUAGE:C,CXX,OBJC,OBJCXX>:-target;${target}>)
+    # We always link as cxx language today using clang so we do not need to
```

Comment:
This comment I think is out of date. I am going to remove it. We are not always linking using cxx, but we are always using cxx to link mixed c/c++/swift content. So I should conditionalize this to be guarded by <COMPILE_LANGUAGE:C,CXX,OBJC,OBJCXX>